### ISSUE TITLE
Missing cosmosdb graph resource even though the azure cli is the same for mongodb

### ISSUE THREAD

**Issue by thomasmillergb** [terraform-providers/terraform-provider-azurerm#3692]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I am trying to create an azure comosdb graph db and collectio. Creating the account is fine  as part of [pr](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1424).

However there is nothing on the terraform docs for the db and collection creation([If there is i cannot see it].(https://www.terraform.io/docs/providers/azurerm/r/cosmosdb_mongo_collection.html#)) about how to actually create the database and graph itself. Even though the az-cli to create [graph db and collection](https://docs.microsoft.com/en-us/azure/cosmos-db/scripts/create-gremlin-database-account-cli) is the exact same for [mongdb](https://docs.microsoft.com/en-us/azure/cosmos-db/scripts/create-mongodb-database-account-cli)

I am wanting the following from az-cli to be integrated into terraform. (This is the exact same as mongodb)
```
az cosmosdb database create \
    --resource-group $resourceGroupName \
    --name $accountName \
    --db-name $databaseName

az cosmosdb collection create \
--collection-name $graphName \
--name $accountName \
--db-name $databaseName \
--resource-group $resourceGroupName \
--partition-key-path /mypartitionkey \
--throughput 1000
``` 

### New or Affected Resource(s)

New Resource
* azurerm_cosmosdb_databse  (like mongodb)
* azurerm_cosmosdb_collection (like mongodb)

### Potential Terraform Configuration
The following should be nearly the same as Mongodb or Mongodb should be intergrated into a new resource which takes care of both graph and Mongodb 

```hcl
data "azurerm_cosmosdb_account" "example" {
  name                = "tfex-cosmosdb-account"
  resource_group_name = "tfex-cosmosdb-account-rg"
}

resource "azurerm_cosmosdb_database" "example" {
  name                = "tfex-cosmos-db"
  resource_group_name = "${data.azurerm_cosmosdb_account.example.resource_group_name}"
  account_name        = "${data.azurerm_cosmosdb_account.example.name}"
}

resource "azurerm_cosmosdb_collection" "example" {
  name = "my-new-graph"
  account-name = "${data.azurerm_cosmosdb_account.example.name}"
  db-name = "${azurerm_cosmosdb_database.example.name}"
  resource-group = "${data.azurerm_cosmosdb_account.example.resource_group_name}"
  partition-key-path = "mypartitionkey"
  throughput = "400"
}


```

### References

[Terraform Doc for Cosmos](https://www.terraform.io/docs/providers/azurerm/r/cosmosdb_mongo_collection.html)
[AZ CLI](https://docs.microsoft.com/en-us/azure/cosmos-db/scripts/create-gremlin-database-account-cli)

* #0000

**Comment by SebRosander** [comment]
Started with the database part. Test seems fine, have to do some documentation and await some feedback on the pull request. I'll try to take a look into the gremlin graph parts if I find some time when this one is finished.

**Comment by SebRosander** [comment]
I've started to look into azurerm_cosmosdb_gremlin_graph parts as well. Still a newbie in go and haven't worked with gremlin that much, so it may take some time. Bear with me.

**Comment by katbyte** [comment]
fixed by #5301
